pkg/services/thumbs/service.go

```diff
@@ -209,13 +213,68 @@ func (hs *thumbService) UpdateThumbnailState(c *models.ReqContext) {
 	c.JSON(http.StatusOK, map[string]string{"success": "true"})
 }
 
+func getDatasourceUIDs(resp *backend.DataResponse) ([]string, error) {
+	if resp == nil {
+		return nil, errors.New("nil response")
+	}
+
+	if resp.Error != nil {
+		return nil, resp.Error
+	}
+
+	if len(resp.Frames) == 0 {
+		return nil, errors.New("empty response")
+	}
+
+	frame := resp.Frames[0]
+	field, idx := frame.FieldByName("ds_uid")
+
+	if field.Len() == 0 || idx == -1 {
+		return nil, errors.New("no ds_uid field")
+	}
+
+	rawValue, ok := field.At(0).(*json.RawMessage)
+	if !ok || rawValue == nil {
+		return nil, errors.New("invalid value in ds_uid field")
+	}
+
+	jsonValue, err := rawValue.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var uids []string
+	err = json.Unmarshal(jsonValue, &uids)
+	if err != nil {
+		return nil, err
+	}
+
+	return uids, nil
+}
+
 func (hs *thumbService) GetImage(c *models.ReqContext) {
 	req := hs.parseImageReq(c, false)
 	if req == nil {
 		return // already returned value
 	}
 
-	res, err := hs.thumbnailRepo.getThumbnail(c.Req.Context(), models.DashboardThumbnailMeta{
+	ctx := c.Req.Context()
+
+	dashQueryResponse := hs.searchService.DoDashboardQuery(ctx, &backend.User{
+		Login: c.SignedInUser.Login,
+		Name:  c.SignedInUser.Name,
+		Email: c.SignedInUser.Email,
+		Role:  string(c.SignedInUser.OrgRole),
+	}, req.OrgID, searchV2.DashboardQuery{
+		UIDs: []string{req.UID},
+	})
+
+	dsUids, err := getDatasourceUIDs(dashQueryResponse)
```

Comment:
So in this place we expect all datasource IDs for a dashboard user can see? How they will be used then? Sorry - trying to understand a context, seems I missed it.

Reply:
if we had the `permissions` field, we would simply check whether the user can query all datasources, so:

```
        {
          "self": {
            "edit": true,
            "view": true,
            "delete": true
          },
          "links": {
            "PD8C576611E62080A": {
              "query": true,
              "view": true
            },
            "some-datasource-uid": {
              "query": true,
              "view": true
            }
          }
        }
```

====> OK


```
        {
          "self": {
            "edit": true,
            "view": true,
            "delete": true
          },
          "links": {
            "PD8C576611E62080A": {
              "query": false,
              "view": true
            },
            "some-datasource-uid": {
              "query": true,
              "view": true
            }
          }
        }
```

====> NOT OK

we won't need to look at the `ds_uid` field at all